wagtail/blocks/field_block.py

```diff
@@ -530,40 +530,57 @@ def _get_callable_choices(self, choices, blank_choice=True):
         choices list with the addition of a blank choice (if blank_choice=True and one does not
         already exist).
         """
+        return self.choices_callable(choices, blank_choice)
```

Comment:
This doesn't seem right... the old version of this method defined `choices_callable` (a function that takes no arguments and returns a list when called), and returns that callable. Here, you've made `choices_callable` into a method, and you're calling that method immediately - which means that `_get_callable_choices` will now return a list, rather than a callable.

I suspect that this change will cause a callable `choices` argument to be evaluated once on initialisation, rather than being evaluated each time the widget is rendered, as we want.

Reply:
I'll verify that and make proper modifications. Callables are tricky to serialize, so method has to be refactored a little bit anyway (so callable isn't nested in a method)